### PR TITLE
Heatmap TODO stubs

### PR DESCRIPTION
**Pull request by rabstejnek**
Implement all TODO stubs and remove them

### CODE REVIEW

File: tests/apps/epi/test_epi_api.py [hunk 1]
```diff
@@ -10,6 +10,25 @@
 
 @pytest.mark.django_db
 class TestEpiAssessmentViewset:
+    def _test_heatmap(self, rewrite_data_files: bool, slug: str, key: int, unpublished: bool):
```
Comment (shapiromatron) [inline tests/apps/epi/test_epi_api.py:13]: this code is almost identical to `test_full_export` - can all 3 tests be consolidated like  `_test_animal_export` in `test_animal_api.py.`

File: tests/apps/lit/test_lit_api.py [hunk 2]
```diff
@@ -99,6 +99,25 @@ def test_reference_year_histogram(self, db_keys):
         resp = c.get(url).json()
         assert resp["data"][0]["type"] == "histogram"
 
+    def test_tag_heatmap(self, rewrite_data_files: str, db_keys):
```
Comment (shapiromatron) [inline tests/apps/lit/test_lit_api.py:102]: like above, can a helper function be written to test `test_references_download` and this function?

File: tests/apps/animal/test_animal_api.py [hunk 3]
```diff
@@ -25,6 +25,25 @@ def _test_animal_export(self, rewrite_data_files: bool, slug: str, key: int):
 
         assert data == json.loads(fn.read_text())
 
+    def _test_heatmap(self, rewrite_data_files: bool, slug: str, key: int, unpublished: bool):
```
Comment (shapiromatron) [inline tests/apps/animal/test_animal_api.py:28]: can you refactor `test_endpoint_export` so we dont have two helper functions that basically do the same thing?

**Comment by rabstejnek** [comment]
# Modal button overhaul
image
image

A regular button shows up for single result, which on click opens up the modal. A dropdown button shows up for multiple results, which on click drops a list of the unique values that the modal targets. A disabled button shows up if there are too many results (more than 10) or no results; there is a mouseover that specifies which it is, and the "too many results" shows an open eye while the "no results" shows a close eye icon.

**Pull request merged**